code/backend/batch/utilities/helpers/embedders/push_embedder.py

```diff
@@ -19,30 +21,48 @@
 
 
 class PushEmbedder(EmbedderBase):
-    def __init__(self, blob_client: AzureBlobStorageClient):
+    def __init__(self, blob_client: AzureBlobStorageClient, env_helper: EnvHelper):
         self.llm_helper = LLMHelper()
         self.azure_search_helper = AzureSearchHelper()
+        self.azure_computer_vision_client = AzureComputerVisionClient(env_helper)
         self.document_loading = DocumentLoading()
         self.document_chunking = DocumentChunking()
         self.blob_client = blob_client
-        config = ConfigHelper.get_active_config_or_default()
+        self.config = ConfigHelper.get_active_config_or_default()
         self.embedding_configs = {}
-        for processor in config.document_processors:
+        for processor in self.config.document_processors:
             ext = processor.document_type.lower()
             self.embedding_configs[ext] = processor
 
     def embed_file(self, source_url: str, file_name: str):
         file_extension = file_name.split(".")[-1]
         embedding_config = self.embedding_configs.get(file_extension)
-        self.__embed(source_url=source_url, embedding_config=embedding_config)
+        self.__embed(
+            source_url=source_url,
+            file_extension=file_extension,
+            embedding_config=embedding_config,
+        )
         if file_extension != "url":
             self.blob_client.upsert_blob_metadata(
                 file_name, {"embeddings_added": "true"}
             )
 
-    def __embed(self, source_url: str, embedding_config: EmbeddingConfig):
+    def __embed(
+        self, source_url: str, file_extension: str, embedding_config: EmbeddingConfig
+    ):
         documents_to_upload: List[SourceDocument] = []
-        if not embedding_config.use_advanced_image_processing:
+        if (
+            embedding_config.use_advanced_image_processing
+            and file_extension
+            in self.config.get_advanced_image_processing_image_types()
+        ):
+            logger.warning("Advanced image processing is not supported yet")
+            image_vectors = self.azure_computer_vision_client.vectorize_image(
+                source_url
+            )
+            logger.info("Image vectors: " + str(image_vectors))
+            # Coming soon, storing the image embeddings in Azure Search
```

Comment:
a todo ?

Reply:
Only put a comment here as I thought it would look odd just logging and doing nothing. I didn't put a TODO as we have it as a task on our issue and it's not something anyone else needs to look at.